src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -461,6 +461,8 @@ public virtual bool IsImplicitClass
             }
         }
 
+        internal virtual bool IsRecord => false;
```

Comment:
Do we have scenarios involving this API on generic/substituted types? Consider other leaf types (retargeting) as follow-up as well.